pkg/store/kotsstore/downstream_store.go

```diff
@@ -423,7 +426,7 @@ func (s *KOTSStore) GetDownstreamVersions(appID string, clusterID string, downlo
 		if err := s.AddDownstreamVersionDetails(appID, clusterID, v, false); err != nil {
 			return nil, errors.Wrap(err, "failed to add details to latest downloaded version")
 		}
-		v.IsDeployable, v.NonDeployableCause = isAppVersionDeployable(v, result, license.Spec.IsSemverRequired)
+		v.IsDeployable, v.NonDeployableCause = isAppVersionDeployable(appID, v, result, license.Spec.IsSemverRequired, nil, nil)
```

Comment:
this means this new EC config comparison logic would not apply to the latest available version. why not keep the previous logic you had but just keep the `break`?

Reply:
how can you rollback to the latest available version?